src/gui/guiFormSpecMenu.cpp

```diff
@@ -3172,6 +3165,14 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 	gui::IGUIFont *old_font = skin->getFont();
 	skin->setFont(m_font);
 
+	// Add a new element that will hold all the background elements as its children
+	// because it is the first added element, all backgrounds will be behind all
+	// the other elements
+	core::rect<s32> background_parent_rect(0, 0, 100000, 100000);
```

Comment:
This kinda looks like the cause for the bug sfan described. The numbers seem arbitrary, mustn't formspec size be factored in here?